packages/hdwallet-provider/src/index.ts

```diff
@@ -37,13 +58,15 @@ class HDWalletProvider {
   public engine: ProviderEngine;
 
   constructor(
-    mnemonic: string | string[],
+    mnemonic: string | string[] | ConstructorOptions,
     provider: string | any,
     addressIndex: number = 0,
     numAddresses: number = 10,
```

Comment:
let's change this to be `numberOfAddresses` as part of my crusade to unabbreviate things ;)